src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',' || 'null') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let joined = '';
+
+    let newSeparator = separator;
+
+    if (separator === null) {
+      newSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        joined += this[i] || '';
+        break;
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        joined += '' + newSeparator;
+        continue;
+      }
+      joined += this[i] + newSeparator;
```

Comment:
I guess it's better to write condition something like this:

      if (this[i] !== undefined && this[i] !== null) {
        joined += this[i];
      }

      if (i !== this.length - 1) {
        joined += separator;
      }
      
this way it looks more clear and compact